protocol/errors.json

```diff
@@ -61,6 +61,26 @@
 	"50002": "internal connection error",
 	"50003": "timeout error",
 
+	/* frontend connection-related */
+	"55000": "internal error (uncategorised)",
+
+	/* frontend connection-related */
+	"55100": "internal error (generic connection error)",
+
+	/* frontend channel-related */
+	"55200": "internal error (generic channel or attachment error)",
+	"55201": "internal error (channel rpc error)",
+
+	/* core channel-related */
+	"55300": "internal error (generic message processing error)",
+	"55301": "internal error (core rpc error)",
+
+	/* core webhook/reactor-related */
+	"55400": "internal error (generic reactor error)",
+
+	/* persistence-related */
```

Comment:
> Should we also include specific error codes for frontends & cores for siege mode i.e. busy?

Initially I wasn't expecting these to be 500 errors tbh; other rate-limiting errors are 4xx. But I think the existing siege mode functionality we added has 500.